charts/node-ipam-controller/templates/deployment.yaml

```
@@ -5,7 +5,14 @@ metadata:
   labels:
     {{- include "node-ipam-controller.labels" . | nindent 4 }}
 spec:
+  {{- if gt (int .Values.replicaCount) 1 }}
+    {{- if not .Values.leaderElection.enabled }}
+      {{- fail "You must set leaderElection.enabled to true to use more than 1 replica" }}
+    {{- end }}
   replicas: {{ .Values.replicaCount }}
+  {{- else }}
+  replicas: 1
+  {{- end }}
```

```suggestion
  {{- end }}
  replicas: {{ .Values.replicaCount }}  
```

In case someone wants to disable the controller and have leader election false, setting `replicas: 0` won't currently work. I.e. try to set in values.yaml `replicaCount: 0` and `leaderElection: false` and run helm template command. The result will always be `replicas: 1`.
This is highly unlikely, because if the controller is in charge of the IPs distribution when it is disabled - nothing will work. Still, I'd prefer to have the exact number of replicas in deployment I specified or an error.

---

@mneverov I think we can use one more condition here like this, wdyt?

```
  {{- else if eq (int .Values.replicaCount) 0 }}
  replicas: 0
```

---

I added it one more condition; I think it looks better now.

---

nope I think your approach makes more sense, sorry for my confusion. fixed it.